Set: Implement `toString` and `fromString` methods

kchulj opened the pull request:
As seen in #2498, `Set!T` is formatted as an `AA`. This is both hard to read and extremely wasteful.
Now, `Set` is formatted as an array, using the updated `toString` and newly implemented `fromString` methods.

e.g. a `Set!string` such as `"_set":{"Hello":true,"world":true}`
will now be formatted as `["Hello", "world"]`.

Fixes #2499

Geod24 commented on source/agora/common/Set.d:
```diff
@@ -236,6 +263,28 @@ unittest
+unittest
+{    
+    auto old_set = Set!uint.from([1, 3, 5, 7, 9]);
+    auto str = old_set.toString();
```
The actual value should be tested, because those will pass as long as they are symmetric.

kchulj replied:
Added a check to test the actual value.

Geod24 commented on source/agora/common/Set.d:
```diff
@@ -51,11 +56,33 @@ public struct Set (T)
+        import std.format : formattedWrite;
 
-        return format("%(%s, %)",
-                      this._set.byKeyValue.filter!(tup => tup.value)
-                      .map!(tup => tup.key));
+        formattedWrite(sink, "[%(%s, %)]", this._set.byKeyValue().filter!(
+            kv => kv.value).map!(kv => kv.key));
```
Why is one side using a policy and the other side is not ?

kchulj replied:
The `deserializeWithPolicy` method requires a policy as a parameter, but the contrary `serializeWithPolicy`  method was found to be not necessary for the `toString`.

Should it be used here?

Commit:
Set: Implement toString and fromString methods

As seen in #2498, Set!T is formatted as an AA. This is both hard
to read and extremely wasteful. Now, the Set is formatted as an
array, using the updated toString and newly implemented fromString
methods.

Fixes #2499

kchulj commented:
Rebased to the latest commit.

hewison-chris reviewed (commented):
LGTM
